edison/static/js/policy.controller.js

```diff
@@ -0,0 +1,150 @@
+
+    var element = document.createElement(type);
+
+    if (classSet != '') {
+        element.setAttribute("class", classSet);
+    }
+    if (id != '') {
+        element.setAttribute("id", id);
+    }
+
+    if (style != '') {
+        element.setAttribute("style", style);
+    }
+
+    return element;
+}
+
+function setCondistion() {
+
+
+    document.getElementById('condition').addEventListener('click', function () {
+
+        var conditionValue = document.getElementById('condition').value;
+        var value = document.getElementById('showValue');
+        var from = document.getElementById('showFrom');
+        var to = document.getElementById('showTo');
+        var lableValue = document.getElementById('lableValue');
+
+        if (conditionValue == '=') {
+            from.style.display = 'none';
+            to.style.display = 'none';
+            lableValue.innerHTML = 'Equal';
+            value.style.display = 'inline';
+
+        }
+        else if (conditionValue == '>' || conditionValue == '<') {
+
+            from.style.display = 'none';
+            to.style.display = 'none';
+            if (conditionValue == '>') {
+                lableValue.innerHTML = 'Above';
+            }
+            else {
+                lableValue.innerHTML = 'Under';
+            }
+            value.style.display = 'inline';
+        }
+        else if (conditionValue == 'Between') {
+
+            from.style.display = 'inline';
+            to.style.display = 'inline';
+            value.style.display = 'none';
+        }
+        else {
+            from.style.display = 'none';
+            to.style.display = 'none';
+            value.style.display = 'none';
+        }
+
+    });
+}
+
+function addCondition(policy) {
+
+    var conditionValue = document.getElementById('condition').value;
+    var condition = document.getElementById('sensor').value + ' ';
+
+    condition = condition + conditionValue + ' ';
+
+
+    if (conditionValue == 'Between') {
+        condition = condition + document.getElementById('from').value + ' ';
+        condition = condition + document.getElementById('to').value;
+    }
+    else {
+
+        condition = condition + document.getElementById('value').value;
+    }
+
+    policy.addCondition(condition);
+
+}
+
+
+function setCommnad(policy) {
+
+    var light = document.getElementById('light').value;
+    var ac = document.getElementById('ac').value;
+    var shutters = document.getElementById('shutters').value;
+
+    policy.addCommandToPolicy(ac, light, shutters);
+
+}
+
+function showCondition(policy, countCondition, elementID) {
+
+    var elemet = document.getElementById(elementID);
+
+    var option = createInitElement('option');
+
+
+    var arrCondotion = policy.condition.split(', ');
+    var sensorValue = arrCondotion[countCondition];
+    option.innerHTML = countCondition + 1 + ': ' + sensorValue;
+
+    elemet.appendChild(option);
+
+}
+
+function addPolicy() {
```

Comment:
I think you should name this function differently because it doesn't add a new policy.

Reply:
this function added a new policy in the end off the function, (sending to the backend), what are you suggest?

Reply:
What I meant is that you're not actively adding a new policy in this function.
What you did here is to add event listeners that one of them adds a new policy when a click occurs.
The function name is missleading and you should be clearer with what the function actually does. You can name it "setEventListeners" for example.

Additionally, it would be better not to pass functions inline to the event listeners, but to extract those functions outside and name them according to what they do.
For example, look at this code:

```
// This is how you did it.
function setEventListeners() {
  document.getElementById('element').addEventListener('click', function () {
    console.log("hello");
  });
}

// This is what I suggest.
function setEventListeners() {
  document.getElementById('element').addEventListener('click', logHello);
}

function logHello() {
  console.log("hello");
}
```

Reply:
OK